internal/clients/team3/clientiigo.go

```diff
@@ -75,3 +99,55 @@ func (c *client) ReceiveCommunication(sender shared.ClientID, data map[shared.Co
 		}
 	}
 }
+
+func (c *client) RuleProposal() string {
+	c.locationService.syncGameState(c.ServerReadHandle.GetGameState())
+	c.locationService.syncTrustScore(c.trustScore)
+	// Will fix properly later
+	shortestSoFar := 999999.0
+	selectedRule := ""
+	for _, rule := range rules.AvailableRules {
+		idealLoc, valid := c.locationService.checkIfIdealLocationAvailable(rule)
+		if valid {
+			ruleDynamics := dynamics.BuildAllDynamics(rule, rule.AuxiliaryVector)
+			distance := dynamics.GetDistanceToSubspace(ruleDynamics, idealLoc)
+			if distance != -1 {
+				if shortestSoFar > distance {
+					if _, ok := rules.RulesInPlay[rule.RuleName]; !ok {
+						shortestSoFar = distance
+						selectedRule = rule.RuleName
+					}
+				}
+			}
+		}
+	}
+	return selectedRule
+}
+
+func (c *client) GetVoteForRule(ruleName string) bool {
+
+	newRulesInPlay := make(map[string]rules.RuleMatrix)
+
+	for key, value := range rules.RulesInPlay {
+		newRulesInPlay[key] = value
+	}
+
+	if _, ok := rules.RulesInPlay[ruleName]; ok {
+		delete(newRulesInPlay, ruleName)
+	} else {
+		newRulesInPlay[ruleName] = rules.AvailableRules[ruleName]
+	}
+
```

Comment:
scratch what i was saying earlier, this does make sense!

Reply:
thanks P :)